BeagleBone_black/src/sensors/mpu9250.cpp

```diff
@@ -301,16 +303,25 @@ void MPU9250::calibrateSensors()
 bool MPU9250::whoAmI()
 {
   uint8_t data;
+  int send_counter;
+
+  for (send_counter = 0; send_counter < 3; send_counter++) {
+    // Who am I checks what address the sensor is at
+    readByte(kWhoAmIMpu9250, &data);
+    if (data != kWhoAmIResetValue1 && data != kWhoAmIResetValue2) {
+      log_.DBG1("MPU9250", "Cannot initialise. Who am I is incorrect");
+      is_online_ = false;
+      Thread::yield();
+    }
+    log_.INFO("MPU9250", "IMU connected to SPI");
+    is_online_ = true;
+    break;
```

Comment:
Why do you have a `break` here? This loop always executes exactly once. Did you mean to have `continue;` at the end of the `if` above?

Reply:
The current implementation always sets `is_online_ = true`